src/main/Config/AppConfig.java

```diff
@@ -105,14 +113,58 @@ public static Javalin appFactory(DeploymentLevel deploymentLevel) {
     app.post("/get-all-activities", activityController.findMyActivities);
 
     /* --------------- PRODUCTION API --------------- */
+
+    // TODO use Access manager for this instead https://javalin.io/documentation#access-manager
+    app.before("/organizations", ctx -> {
+      String sessionUsername = ctx.sessionAttribute("username");
+      var sessionUser = userDao.get(sessionUsername);
+
+      if (sessionUsername == null || sessionUsername.isEmpty() || sessionUser.isEmpty()) {
+        throw new HttpResponseException(401, "Authentication failed", new HashMap<>());
+      } else if (sessionUser.get().getUserType() != UserType.Developer) {
+        throw new HttpResponseException(403, "Insufficient permissions", new HashMap<>());
+      }
+    });
+
     app.get("/organizations", productionController.readAllOrgs);
     app.post("/organizations", productionController.createOrg);
+
+    app.before("/organizations/:orgId", ctx -> {
```

Comment:
what does app.before do?

Reply:
It's middleware that will execute prior to the specific route handlers for any request matching, in this case, `"/organizations/:orgId"`. This makes it easy to add auth and common logic for every handler in one place